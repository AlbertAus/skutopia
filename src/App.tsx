import { useState } from "react";
import "./App.css";
import OrdersOutput from "./containers/orderContainer";
function App() {
  const [endPoint, setEndPoint] = useState("inventory");
  const [source, setSource] = useState("shopify");
  return (
    <div className="App">
      {/* setup End Point system */}
      <div>
        <h1>Choose end point</h1>
        <button id='inventory' onClick={() => setEndPoint("inventory")}>Inventory</button>
        <button id='warehouse' onClick={() => setEndPoint("warehouse")}>Warehouse </button>
      </div>
      {/* setup source */}
      <div>
        <h1>Choose integration source</h1>
        <button onClick={() => setSource("shopify")}>shopify</button>
        <button onClick={() => setSource("woo_commerce")}>woo_commerce </button>
      </div>
      <h2 className="App-menu">&#60; RETURN TO ORDERS</h2>

      <OrdersOutput endPoint={endPoint} source={source} />
    </div>
  );
}

export default App;
