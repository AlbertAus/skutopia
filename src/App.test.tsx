import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import App from './App';

Enzyme.configure({ adapter: new Adapter() });

describe('Test App component', () => {
  it('case: expect to match snapshot', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
});

  it('case: expect render correctly', () => {
      const wrapper = shallow(<App />);

      expect(wrapper.find('.App').length).toEqual(1);
  });

  it('case: expect button inventory to be 1', () => {
    const wrapper = mount(<App />);

    expect(wrapper.find('button[id="inventory"]').length).toEqual(1);
});
});