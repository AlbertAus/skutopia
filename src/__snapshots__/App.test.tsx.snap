// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component case: expect to match snapshot 1`] = `ShallowWrapper {}`;
